reliability-engineering/terraform/modules/concourse-secrets-admin/iam.tf

```diff
@@ -0,0 +1,87 @@
+resource "aws_iam_role" "concourse_secrets_admin" {
+  name = "${var.deployment}-${var.concourse_team_name}-concourse-secrets-admin"
+
+  assume_role_policy = <<-EOF
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Effect": "Allow",
+        "Principal": {
+          "Federated": "${module.concourse_base.aws_iam_openid_connect_provider_arn}"
+        },
+        "Action": "sts:AssumeRoleWithWebIdentity",
+        "Condition": {
+          "StringEquals": {
+            "${var.oidc_host_path}:aud": "${var.github_oauth_client_id}",
+            "aws:RequestTag/t${var.trusted_github_team_id}": "t"      // instantiated in team.tf files in tech-ops-private
+          }                                                          // and we need to supply github team id as a discrete no.
+        }                                                           // change name to team_name
+      },                                                            // VPN cidr's (IP restrictions)
+      {
+        "Sid": "AllowPassSessionTagsAndTransitive",
+        "Effect": "Allow",
+        "Action": "sts:TagSession",
+        "Principal": {
+          "Federated": "${module.concourse_base.aws_iam_openid_connect_provider_arn}"
+        }
+      }
+    ]
+  }
+EOF
+}
+
+resource "aws_iam_role_policy" "concourse_secrets_admin" {
+  name = "${var.deployment}-${var.concourse_team_name}-concourse-secrets-admin-policy"
+  role = aws_iam_role.concourse_secrets_admin.id
+
+  policy = <<-POLICY
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Action": [
+          "ssm:GetParameter",
+          "ssm:GetParameters",
+          "ssm:GetParametersByPath"
+        ],
+        "Effect": "Allow",
+        "Resource": [
+          "arn:aws:ssm:eu-west-2:${data.aws_caller_identity.account.account_id}:parameter/${var.deployment}/concourse/worker/${var.concourse_team_name}/*"
+        ]
+      }, {
+        "Action": [
+          "ssm:GetParameter",
+          "ssm:GetParameterHistory",
+          "ssm:GetParameters",
+          "ssm:GetParametersByPath",
+          "ssm:DeleteParameter",
+          "ssm:PutParameter"
+        ],
+        "Effect": "Allow",
+        "Resource": [
+          "arn:aws:ssm:eu-west-2:${data.aws_caller_identity.account.account_id}:parameter/${var.deployment}/concourse/pipelines/${var.concourse_team_name}/*"
+        ]
+      }, {
+        "Action": [
+          "ssm:DeleteParameter",
+          "ssm:PutParameter"
+        ],
+        "Effect": "Deny",
+        "Resource": [
+          "arn:aws:ssm:eu-west-2:${data.aws_caller_identity.account.account_id}:parameter/${var.deployment}/concourse/pipelines/${var.concourse_team_name}/readonly_*"
```

Comment:
I think the values at the moment are camel case rather than _'s

Reply:
Do you mean the values of one or more of the variables used here?

Reply:
We already have a few read only values within the SSM Storage, and they are prefixed `readonlyFooBar` for example `/cd-dev/concourse/pipelines/test/readonlySecretsKmsKeyId`

Reply:
All the ones I can see are underscored.

Reply:
Hmm, OK. This was copied from the existing policy [1] that covers access from the worker machines, so if there is an issue with the way the policy matches the parameters, it's an existing one.

1: https://github.com/alphagov/tech-ops/blob/master/reliability-engineering/terraform/modules/concourse-worker-pool/iam.tf#L54

This is one to investigate further with some concrete examples.